parcels/_index_search.py

```diff
@@ -0,0 +1,336 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from parcels._typing import (
+    GridIndexingType,
+    InterpMethodOption,
+)
+from parcels.tools.statuscodes import (
+    FieldOutOfBoundError,
+    FieldOutOfBoundSurfaceError,
+    _raise_field_out_of_bound_error,
+    _raise_field_out_of_bound_surface_error,
+    _raise_field_sampling_error,
+)
+
+from .grid import GridType
+
+if TYPE_CHECKING:
+    from .field import Field
+    from .grid import Grid
+
+
+def search_indices_vertical_z(grid: Grid, gridindexingtype: GridIndexingType, z: float):
+    if grid.depth[-1] > grid.depth[0]:
+        if z < grid.depth[0]:
+            # Since MOM5 is indexed at cell bottom, allow z at depth[0] - dz where dz = (depth[1] - depth[0])
+            if gridindexingtype == "mom5" and z > 2 * grid.depth[0] - grid.depth[1]:
+                return (-1, z / grid.depth[0])
+            else:
+                _raise_field_out_of_bound_surface_error(z, 0, 0)
```

Comment:
It's not ideal that this function passes `y=0` and `x=0`. I understand we don't know what `x` and `y` are at this moment, but printing them as zeros can be misleading to users. Would it not be better to print `not available` or `unknown` or so? Or is there no way to somehow figure out what `x` and `y` are?

Reply:
The previous code was also putting them to be 0, so this was just a continuation of that. Easy fix to set `x` and `y` to be `None`!

Reply:
OK, and could we then print these `None`s as 'unknown' in the actual warning message to the user? Would then make more sense to them?